modules/core/shared/src/main/scala/io/circe/Decoder.scala

```diff
@@ -604,7 +604,7 @@ object Decoder
    *
    * @group Decoding
    */
-  implicit final val decodeJavaBoolean: Decoder[java.lang.Boolean] = decodeBoolean.map(java.lang.Boolean.valueOf)
+  implicit final lazy val decodeJavaBoolean: Decoder[java.lang.Boolean] = decodeBoolean.map(java.lang.Boolean.valueOf)
```

Comment:
Is there a reason to make these lazy? There is some overhead, and since as far as I know all of these non-`java.time` Java types are fully supported by Scala.js, I'd rather not make this change where it's not necessary.

Reply:
I was thinking that making these lazy means that Scala.JS can remove its `java.lang.*` shims from output if otherwise unused, resulting in a smaller output JS size. I can revert this though if you like, no prob.

Reply:
Hmm, we should verify that, but sounds reasonable. I'm merging this now, but will change the non `java.lang` ones back in a follow-up.